src/index.js

```diff
@@ -22,41 +22,35 @@ const createGetData = fetch => async (url, opts) => {
   const embed = parse(text)
 
   const scripts = embed
-    .filter(e => e.tagName === 'html')[0]
-    .children.filter(e => e.tagName === 'body')[0]
-    .children.filter(e => e.tagName === 'script')
+    .find(el => el.tagName === 'html')
+    .children.find(el => el.tagName === 'body')
+    .children.filter(({ tagName }) => tagName === 'script')
 
-  const resourceScript = scripts.filter(
-    e => e.attributes.findIndex(a => a.value === 'resource') !== -1
+  let script = scripts.find(script =>
+    script.attributes.some(({ value }) => value === 'resource')
   )
 
-  if (resourceScript.length > 0) {
+  if (script !== undefined) {
     // found data in the older embed style
-    return JSON.parse(decodeURIComponent(resourceScript[0].children[0].content))
+    return normalizeData({
+      data: JSON.parse(decodeURIComponent(script.children[0].content))
+    })
   }
 
-  const hydrateScript = scripts.filter(
-    e => e.children[0] && /%22data%22%|"data":/.test(e.children[0].content)
+  script = scripts.find(script =>
+    script.attributes.some(({ value }) => value === 'initial-state')
   )
 
-  if (hydrateScript.length > 0) {
-    // found hydration data
-    // parsing via looking for { to be a little bit resistant to code changes
-    const scriptContent = hydrateScript[0].children[0].content.includes(
-      '%22data%22%'
-    )
-      ? decodeURIComponent(hydrateScript[0].children[0].content)
-      : hydrateScript[0].children[0].content
-    return normalizeData(
-      JSON.parse(
-        '{' +
-          scriptContent
-            .split('{')
-            .slice(1)
-            .join('{')
-            .trim()
-      )
-    )
+  if (script !== undefined) {
+    // found data in the new embed style
+    const data = JSON.parse(
+      Buffer.from(decodeURIComponent(script.children[0].content), 'base64')
+    ).data.entity
+    // they removed/renamed some things, which for backwards compatibility we need to add back
+    data.external_urls = { spotify: spotifyURI.formatOpenURL(data.uri) }
+    data.release_date = data.releaseDate.isoString
+    data.audio = data.audioPreview.url
```

Comment:
@Eithoo why this backward compatibility is necessary?

Reply:
Because they were returned by `getData` previously and should be there until the new major update (4.0.0) - someone might be using it in their project. I guess this fix is a minor/patch update, so everything should be backward compatible
https://semver.org/

Reply:
how do you know `release_date` and `audio` is being there, if isn't any assertion checking the value is there?

Reply:
I will be happy to accept this with the proper test assertions in a separate PR; For now, I'm going to remove these props for move this PR forward